test/ai/utils/enemy-command-utils.ts

```diff
@@ -5,22 +5,51 @@ import { randSeedInt } from "#app/utils/random-utils";
 import type { MoveId } from "#enums/move-id";
 import { vi } from "vitest";
 
+/** The number of move selections calculated by {@linkcode getEnemyMoveChoices} */
 const NUM_TRIALS = 300;
 
-export type MoveChoiceSet = { [key: number]: number };
+/**
+ * A record for the number of times each move in a Pokemon's moveset
+ * is selected across all trials of {@link getEnemyMoveChoices | move choice evaluation}
+ */
+type MoveChoiceSet = Partial<Record<MoveId, number>>;
 
-export function getEnemyMoveChoices(scene: BattleScene, pokemon: EnemyPokemon, moveChoices: MoveChoiceSet): void {
+/**
+ * Creates a {@linkcode MoveChoiceSet} with entries for each of the given
+ * {@linkcode EnemyPokemon}'s moves set to `0`.
+ */
+export function initMoveChoiceSet(pokemon: EnemyPokemon): MoveChoiceSet {
+  const movesetIds = pokemon.getMoveset().map((mv) => mv.moveId);
+  return movesetIds.reduce<MoveChoiceSet>((choiceSet, moveId) => {
+    choiceSet[moveId] = 0;
+    return choiceSet;
+  }, {});
+}
+
+/**
+ * Evaluates and reports an enemy Pokemon's move choice in the current battle state
+ * over {@linkcode NUM_TRIALS} trials.
+ * @param pokemon - The {@linkcode EnemyPokemon} whose move selection is evaluated
+ * @returns A {@linkcode MoveChoiceSet} with the total number of times each of the Pokemon's
+ * moves are selected across all trials
+ */
+export function getEnemyMoveChoices(pokemon: EnemyPokemon): MoveChoiceSet {
   // Use an unseeded random number generator in place of the mocked-out randBattleSeedInt
-  vi.spyOn(scene, "randBattleSeedInt").mockImplementation((range, min?) => {
+  vi.spyOn(pokemon.scene as BattleScene, "randBattleSeedInt").mockImplementation((range, min?) => {
     return randSeedInt(range, min);
   });
+
+  const moveChoices = initMoveChoiceSet(pokemon);
   for (let i = 0; i < NUM_TRIALS; i++) {
     const queuedMove = pokemon.getNextMove();
-    moveChoices[queuedMove.move.id]++;
+    /** @todo Is there a way to show this is defined after {@linkcode initMoveChoiceSet} without the bang? */
+    moveChoices[queuedMove.move.id]!++;
```

Comment:
Wow, I can see where your frustration for this came from.
I came up with this solution:
```suggestion
    const prev =  moveChoices[queuedMove.move.id] ?? 0;
    moveChoices[queuedMove.move.id] = prev + 1;
```